xdsl/dialects/builtin.py

```diff
@@ -2268,6 +2279,35 @@ def create_dense_float(
         else:
             data = cast(Sequence[float], data)
 
+        assert not isinstance(type.element_type, ComplexType)
+        return DenseIntOrFPElementsAttr([type, BytesAttr(type.element_type.pack(data))])
+
+    @overload
+    @staticmethod
+    def create_dense_complex(
+        type: RankedStructure[ComplexType[_IntegerTypeInvT]],
+        data: Sequence[tuple[int, int]],
+    ) -> DenseIntOrFPElementsAttr[ComplexType[_IntegerTypeInvT]]: ...
+
+    @overload
+    @staticmethod
+    def create_dense_complex(
+        type: RankedStructure[ComplexType[_FloatAttrTypeInvT]],
+        data: Sequence[tuple[float, float]],
+    ) -> DenseIntOrFPElementsAttr[ComplexType[_FloatAttrTypeInvT]]: ...
+
+    @overload
+    @staticmethod
+    def create_dense_complex(
+        type: RankedStructure[ComplexType[ComplexElementT]],
+        data: Sequence[tuple[float, float]] | Sequence[tuple[float, float]],
+    ) -> DenseIntOrFPElementsAttr[ComplexType[ComplexElementT]]: ...
+
+    @staticmethod
+    def create_dense_complex(
+        type: RankedStructure[ComplexType],
```

Comment:
```suggestion
        type: RankedStructure[ComplexType[ComplexElementT]],
```
We can also specialise here